src/scenes/ProgressReports/Detail/WorkflowEventList.tsx

```diff
@@ -0,0 +1,68 @@
+import { ArrowForward } from '@mui/icons-material';
+import {
+  Box,
+  Collapse,
+  Divider,
+  Paper,
+  Stack,
+  Typography,
+} from '@mui/material';
+import { Fragment } from 'react';
+import { ProgressReportStatusLabels as StatusLabels } from '~/api/schema/enumLists';
+import { FormattedDate } from '~/components/Formatters';
+import { RichTextView } from '~/components/RichText';
+import { ProgressReportDetailFragment } from './ProgressReportDetail.graphql';
+
+interface WorkFlowEventListProps {
+  report?: ProgressReportDetailFragment;
+  showNotes?: boolean;
+}
+
+export const WorkFlowEventList = ({
+  showNotes,
+  report,
+}: WorkFlowEventListProps) => {
+  if (!report?.workflowEvents) {
+    return null;
+  }
+
+  const statusEvents = [
+    {
+      ...report.workflowEvents[0]!,
+      status: 'NotStarted' as keyof typeof StatusLabels,
+    },
+    ...report.workflowEvents,
```

Comment:
I wasn't sure how to make this implementation work in this component how I have it setup.  I am using the .slice method with the map function to offset the maps event from the StatusEvent array so that on line 49 {StatusLabels[statusEvents[i + 1]!.status]} is not undefined.  If I move the .slice to line 34 the arrays is no longer offset and line 49 throws and error as Undefined because there is not another position in the array with a status property.  Do you know of a better way to handle this?   Andrew agreed that this wouldn't work (at least with my approach) but did help me refactor this and rename things to be more clear.

Reply:
I haven't taken a long look at this yet but I may need you to expand on this a little or possibly zoom or huddle about it at some point.  I can also meet with @andrewmurraydavid if he understands what you mean.